### DISCUSSION TITLE
reference between deck

### DISCUSSION THREAD

**Discussion by chenxizhang** [marp-team/marp#243] [Q&A] [answered]
when I reference a deck(a.md) in another deck (b.md), and then I used `npx marp` to generate the html file, let's saying there will be two files:  a.html, b.html

I want to have the correct link in `b.html`, to reference the `a.html`, not `a.md`

possible?

**Comment by chenxizhang** [comment]
![image](https://user-images.githubusercontent.com/1996954/146936409-91ed35fc-0a8c-4a9c-b2a9-db4599ea0174.png)

**Comment by yhatt** [comment, marked as answer]
For to respect exact intent of the document author, Marp has no implicit replaces for hyperlink URLs when converted Markdown to HTML. Thus, `.md` extension in URL within converted HTML still will become `.md`.

If you have expected slides as published as a bunch of HTML, is there a reason why not set URL as ending with `.html` instead of `.md`?

---

If you really need, Marp CLI can define a custom plugin through `engine.js`.

```javascript
// engine.js
module.exports = ({ marp }) => marp.use((md) => {
  const { normalizeLink } = md

  md.normalizeLink = (...args) => {
    const originalResult = normalizeLink(...args)

    // Replace URL: ".md" -> ".html"
    return originalResult.replace(/\.md$/, '.html')
  }
})
```

```bash
npx marp --engine ./engine.js a.md
```

Marp/Marpit is based on [markdown-it](https://github.com/markdown-it/markdown-it), and [`MarkdownIt#normalizeLink`](https://markdown-it.github.io/markdown-it/#MarkdownIt.prototype.normalizeLink) would be helpful to return normalized URL(s).

**Reply by chenxizhang** [reply]
Thanks Yuki.

The scenario is real, you can imagine that I am writing a book, and want to make every chapter to a deck, I might referent one chapter in another one.  **And I want the link will work on both md and html file**, that mean, when I am in the md file, click the link, will navigate to the md target. when I am in the html file, click the link, will navigate to the html target.

I used `gitbook` for another book before, and the compiler will do this conversion automatocally. 

I used your solution, the plugin workaround, it works great for me. really amazing.
